pallets/identity/src/lib.rs

```diff
@@ -246,7 +240,11 @@ decl_module! {
             let cdd_id = Context::current_identity_or::<Self>(&cdd_key)?;
 
             let cdd_providers = T::CddServiceProviders::get_members();
-            ensure!( cdd_providers.contains(&cdd_id), Error::<T>::UnAuthorizedCddProvider);
+            ensure!(cdd_providers.contains(&cdd_id), Error::<T>::UnAuthorizedCddProvider);
+            T::ProtocolFee::charge_fee(
```

Comment:
`_register_did` function that is called after charging fee does additional checks and can return an error without making state changes.